src/gui.ts

```diff
@@ -0,0 +1,280 @@
+}
+
+function createGameTitleLabel() {
+  return blessed.text({
+    top: 0,
+    left: TEXT_LEFT_OFFSET,
+    content: "The Game of Hex",
+    tags: true,
+  });
+}
+
+function createBoardRightEdge(boardLayout: blessed.Widgets.BoxElement, y: number, gameState: GameState) {
+  boardLayout.append(
+    blessed.text({
+      top: (y + 2) * CELL_HEIGHT,
+      left: (2 + gameState.board.length) * CELL_WIDTH + ((y + 1) * CELL_WIDTH) / 2,
+      content: " \\",
+      tags: true,
+      style: {
+        fg: "black",
+        bg: "gray",
+      },
+    })
+  );
+}
+
+function createBoardLeftEdge(boardLayout: blessed.Widgets.BoxElement, y: number) {
+  boardLayout.append(
+    blessed.text({
+      top: (y + 2) * CELL_HEIGHT,
+      left: CELL_WIDTH + (y * CELL_WIDTH) / 2,
+      content: " \\",
+      tags: true,
+      style: {
+        fg: "black",
+        bg: "gray",
+      },
+    })
+  );
+}
+
+function createRowHeaderLabel(boardLayout: blessed.Widgets.BoxElement, y: number) {
+  boardLayout.append(
+    blessed.text({
+      top: (y + 2) * CELL_HEIGHT,
+      left: (y * CELL_WIDTH) / 2,
+      content: getRowLabelDisplayContent(y),
+      tags: true,
+      style: {
+        fg: "black",
+        bg: "gray",
+      },
+    })
+  );
+}
+
+function createBoardBottomEdge(boardLayout: blessed.Widgets.BoxElement, gameState: GameState, x: number) {
+  boardLayout.append(
+    blessed.text({
+      top: (gameState.board.length + 2) * CELL_HEIGHT,
+      left: (x + 2) * CELL_WIDTH + (gameState.board.length + 1) * CELL_WIDTH / 2,
+      content: " -",
+      tags: true,
+      style: {
+        fg: "white",
+        bg: "gray",
+      },
+    })
+  );
+}
+
+function createBoardTopEdge(boardLayout: blessed.Widgets.BoxElement, x: number) {
+  boardLayout.append(
+    blessed.text({
+      top: CELL_HEIGHT,
+      left: (x + 2) * CELL_WIDTH,
+      content: " -",
+      tags: true,
+      style: {
+        fg: "white",
+        bg: "gray",
+      },
+    })
+  );
+}
+
+function createColumnHeaderLabel(boardLayout: blessed.Widgets.BoxElement, x: number) {
+  boardLayout.append(
+    blessed.text({
+      top: 0,
+      left: (x + 2) * CELL_WIDTH,
+      content: getColLabelDisplayContent(x),
+      tags: true,
+      style: {
+        fg: "black",
+        bg: "gray",
+      },
+    })
+  );
+}
+
+function createBoardOuterLayout(gameState: GameState) {
+  return blessed.box({
+    top: "center",
+    left: "center",
+    width: CELL_WIDTH * (gameState.board.length + 3) * 1.5 + PADDING_WIDTH,
+    height: CELL_HEIGHT * (gameState.board.length + 3) + PADDING_HEIGHT,
+    tags: true,
+    border: {
+      type: "line",
+    },
+    style: {
+      border: {
+        fg: "white",
+      },
+      bg: "gray",
+    },
+  });
+}
+
+function createCurrentPlayerMsg(
+  gameState: GameState
+): blessed.Widgets.TextElement {
+  return blessed.text({
+    top: "center",
+    left: TEXT_LEFT_OFFSET,
+    content: `{bold}${gameState.turn}{/bold} turn to play`,
```

Comment:
Could be in uppercase to be more easy to read ? Or in another color like black/white(but need to change background if i remember well)